tests/data/fields/text_field_test.py

```diff
@@ -88,19 +88,19 @@ def test_index_converts_field_correctly(self):
         t_index = vocab.add_token_to_namespace("t", namespace='characters')
         c_index = vocab.add_token_to_namespace("c", namespace='characters')
 
-        field = TextField(["A", "sentence"], [token_indexers["single id"](token_namespace="words")])
+        field = TextField(["A", "sentence"], [token_indexers["single id"](namespace="words")])
         field.index(vocab)
         # pylint: disable=protected-access
         assert field._indexed_tokens == [[capital_a_index, sentence_index]]
```

Comment:
I was looking at this, and wondered why this still passed.  Then I realized that even though the return value of `as_array` is a `Dict`, `token_indexers` is still a list, and we have the issue of how to map from this list to keys in the `Dict`.  Why don't we just make the `token_indexers` a dict, and the user assigns their own keys?  I think that would clear up the issue with using "words" vs. "tokens", too.  What do you think?